test/fingerprint_opts.c

```diff
@@ -0,0 +1,54 @@
+#include <pg_query.h>
+#include <pg_query_fingerprint.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "fingerprint_opts_tests.c"
+
+int main()
+{
+	size_t i;
+	bool ret_code = 0;
+
+	for (i = 0; i < testsLength; i += 3)
+	{
+		PgQueryParseMode mode = atoi(tests[i+1]);
+		PgQueryFingerprintResult result = pg_query_fingerprint_opts(tests[i], mode);
+
+		if (result.error)
+		{
+			ret_code = -1;
+			printf("%s\n", result.error->message);
+			pg_query_free_fingerprint_result(result);
+			continue;
+		}
+		else if (strcmp(result.fingerprint_str, tests[i + 2]) == 0)
+		{
+			printf(".");
+		}
+		else
+		{
+			ret_code = -1;
+			printf("INVALID result for \"%s\" with %d mode\nexpected: \"%s\"\nactual: \"%s\"\nactual tokens: ", tests[i], mode, tests[i + 2], result.fingerprint_str);
+			pg_query_fingerprint_with_opts(tests[i], mode, true);
```

Comment:
It looks like the `actual tokens: ` should be removed, as well as this `pg_query_fingerprint_with_opts` call. Or did you intend to do something more here?

Reply:
This was copied from the existing fingerprint.c. I believe the second call prints the actual tokens.